src/encoded/reports/metadata.py

```diff
@@ -0,0 +1,380 @@
+                del self.param_list['cart']
+            except KeyError:
+                raise HTTPBadRequest(explanation='Specified cart does not exist.')
+            else:
+                self.param_list['@id'].extend(
+                    cart.get('elements', [])
+                )
+
+    def _maybe_add_json_elements_to_param_list(self):
+        try:
+            self.param_list['@id'].extend(
+                self.request.json.get('elements', [])
+            )
+        except ValueError:
+            pass
+
+    def _get_field_params(self):
+        return [
+            ('field', p)
+            for p in self.param_list.get('field', [])
+        ]
+
+    def _get_at_id_params(self):
+        return [
+            ('@id', p)
+            for p in self.param_list.get('@id', [])
+        ]
+
+    def _get_default_params(self):
+        return self.DEFAULT_PARAMS
+
+    def _build_query_string(self):
+        self.query_string.drop('limit')
+        self.query_string.drop('option')
+        self.query_string.extend(
+            self._get_default_params()
+            + self._get_field_params()
+            + self._get_at_id_params()
+        )
+
+    def _get_search_path(self):
+        return self.SEARCH_PATH
+
+    def _validate_request(self):
+        type_params = self.param_list.get('type', [])
+        if len(type_params) != 1:
+            raise HTTPBadRequest(explanation='URL requires one "type" parameter.')
+        return True
+
+    def _initialize_report(self):
+        self._build_header()
+        self._split_column_and_fields_by_experiment_and_file()
+        self._set_positive_file_param_list()
+
+    def _build_params(self):
+        self._add_fields_to_param_list()
+        self._initialize_at_id_param()
+        self._maybe_add_cart_elements_to_param_list()
+        self._maybe_add_json_elements_to_param_list()
+
+    def _build_new_request(self):
+        self._build_query_string()
+        request = self.query_string.get_request_with_new_query_string()
+        request.path_info = self._get_search_path()
+        request.registry = self.request.registry
+        return request
+
+    def _get_search_results_generator(self):
+        return search_generator(
+            self._build_new_request()
+        )
+
+    def _should_not_report_file(self, file_):
+        conditions = [
+            not file_matches_file_params(file_, self.positive_file_param_list),
+            self.visualizable_only and not is_file_visualizable(file_),
+            self.raw_only and file_.get('assembly'),
+            file_.get('restricted'),
+            file_.get('no_file_available'),
+        ]
+        return any(conditions)
+
+    def _get_experiment_data(self, experiment):
+        return {
+            column: make_experiment_cell(fields, experiment)
+            for column, fields in self.experiment_column_to_fields_mapping.items()
+        }
+
+    def _get_file_data(self, file_):
```

Comment:
I assume "file_" as in "you cannot use 'file' so using 'file_'" ..?

Reply:
Yep.